.github/workflows/ci.yaml

```diff
@@ -0,0 +1,94 @@
+name: CI Tests
+
+on:
+  push:
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
+
+jobs:
+  smoke-test-install:
+    name: Smoke Test Installation
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ["3.9", "3.10", "3.11", "3.12"]
```

Comment:
Upgrade the versions, cover from 3.10 to 3.14 (those are the currently officially supported versions of Python). Please adapt the pyproject.toml to indicate that all those versions are supported.